### PR TITLE
Remaining AC3 documentation updates

### PR DESCRIPTION
**Pull request by hwillson**
This PR houses and tracks the remaining AC3 documentation work that needs to be completed for launch.

**TODO:**

- [ ] Remaining local state updates
    - [x] Handling `@client` fields with field policies
      - [ ] add a mutation example
      - [ ] add a merge function example
    - [x] Using reactive variables to track `@client` state
    - [x] Integrating `@client` into remote queries
    - [ ] Handling async operations with field policies and reactive variables
    - [x] Handling `@client` fields with the cache
    - [x] Using `@client` fields as variables
    - [x] Client side schema
    - [x] Refactor content layout
        - [x] Combine updating/querying local state
        - [x] Adjust all subsections to be l2's to show in right hand menu
        - [x] Deprecate in local resolvers title
- [x] Remove broken Meteor docs
- [x] Consider cache policies vs. field policies specific naming
- [x] Remove all Apollo Component docs examples
- [x] `cache.modify`
  - [x] Be sure to demonstrate both updating and removal
- [x] `cache.identify`
- [x] Migration guide for moving from local resolvers to cache policies
- [ ] `@apollo/client/core`, `@apollo/client/utilities`, `@apollo/client/testing`, `@apollo/client/cache` entry points
- [x] `canRead` and `readField` helpers
- [x] `cache.evict(id, fieldName)` changes to support specific entity field eviction
- [x] `readQuery` / `writeQuery` API updates (e.g. `options.id`)
- [x] Remove all `FragmentMatcher`, `HeuristicFragmentMatcher`, and `IntrospectionFragmentMatcher` references
- [ ] Explain new cache `console.log` warnings, what they mean, how to resolve them
- [x] How to address dangling references
- [ ] Fall back to `cache-first` after `cache-and-network` or `network-only` (8ce2b4f7c8e631a46ed41cd205c34ede38f202c9)
- [ ] Update all `read` / `merge` docs to make sure they reflect the final version of the API
- [x] Replace all mentions of `node-fetch` with `cross-fetch` (fdcd519cddc6fd44b57e8254e5eb13b5f257c2ee)
- [ ] Silencing broadcasting (https://github.com/apollographql/apollo-client/pull/6288)
- [x] Review migration guide to make sure all important/common path breaking changes are mentioned
- [ ] Consider updating the intro path to mutations to replace the new default approach of using `cache.modify` with a `readQuery` / `writeQuery` approach instead (then briefly mention `cache.modify` as a more advanced alternative)
- [x] Make sure `@live` support removal is mentioned somewhere
- [ ] Update the migration guide
  - [ ] Mention local state management changes, including reactive variables
  - [ ] Outline how to migrate from local resolvers to cache policies
  - [ ] Decide if we want to highlight important breaking changes, or refer to the `CHANGELOG` (as we're duplicating details)
- [x] Consider adding a `InMemoryCache` section under `API - CORE` explaining main API cache functions and options
- [x] Make sure all `cacheRedirects` mentions/examples have been replaced with type policies
- [ ] Refactor all pagination documentation
  - [ ] Document new pagination helpers (https://github.com/apollographql/apollo-client/pull/6464 and https://github.com/apollographql/apollo-client/pull/6465)
- [ ] Deprecate `fetchMode`'s `updateQuery`, replacing with field policies (https://github.com/apollographql/apollo-client/pull/6464)
- [x] Remove broken `Performance > Query splitting` section
- [x] Decide if we want to write a new `Performance > Query splitting` section using `read` and `merge` functions
- [x] Replace `cache.makeVar` references to direct `makeVar` imports (https://github.com/apollographql/apollo-client/pull/6512)

### CODE REVIEW

**Comment by hwillson** [comment]
@StephenBarlow @benjamn @jcreighton I'll take this out of draft as I think there's starting to be a bit much to review all at once. I'll continue working on the outstanding todo items in separate PR's off of this one. As far as reviewing goes, commit by commit should be a reasonable path.

One quick thing to note @stephen - I don't think there's any value in reviewing the `Local resolvers (DEPRECATED)` page. I've removed a few sections that are no longer needed, but other than that we're going to be removing local resolvers completely in the next major AC version. Maybe just a quick review to make sure the deprecation warning message is okay?

Oh, and one more thing - Netlify doc previews are currently broken for all AC PR's. I'll follow up with @trevorblades on this.

Thanks all!

File: docs/source/caching/cache-interaction.md [hunk 1]
```diff
@@ -196,6 +196,183 @@ client.writeQuery({
...
+The [`cache.writeQuery`](#writequery-and-writefragment) and [`cache.writeFragment`](#writequery-and-writefragment) methods do a great job of adding data to the cache, but their use can be problematic when trying to remove specific data from a field in the cache. The typical cycle of reading data, modifying it, and writing it back into the cache does not always simply replace the old data, as it may trigger custom [`merge` functions](./cache-field-behavior/#the-merge-function) which attempt to combine incoming data with existing data, leading to confusion.
+
+For cases where you want to apply a specific transformation to an existing field value in the cache, the `cache.modify` function can be helpful. `cache.modify` takes an entity ID and an object mapping field names to modifier functions. For the specified entity, each field modifier function is called with the current value or references of the field, and should return a new value for the field, without modifying the existing value (which is frozen in development).
+
+For example, here's how you might remove a specific `Comment` from a paginated `Thread.comments` array:
+
+```js
+const idToRemove = 'C1';
+
+cache.modify({
```
Comment (henryqdineen) [inline docs/source/caching/cache-interaction.md:263]: If there are multiple `comments` fields in the cache with different args is there a way for `cache.module` to target one of them specifically. Consider this super simple cache extraction where there is a top-level scalar typed `foo` field :

```
{
  "ROOT_QUERY": {
    "__typename": "Query",
    "foo({\"id\":1})": 'bar',
    "foo({\"id\":2})": 'baz'
  }
}
```

I want to update `foo({\"id\":2})` but with `cache.modify()`  I can't seem to target it.
```
cache.modify({
  fields: {
    foo() {
      // this gets called twice
    },
  },
});
```

Is `reactQuery/writeQuery` the way to do this going forward. Thanks! Let me know if this deserves an issue. Thanks!
cc @hwillson @benjamn

Comment (darkbasic) [reply docs/source/caching/cache-interaction.md:263]: @henryqdineen unfortunately not: this is a feature I requested and which looked like was targeted for AC3, but apparently they changed their mind about it :(

Comment (benjamn) [reply docs/source/caching/cache-interaction.md:164]: @henryqdineen See https://github.com/apollographql/apollo-client/issues/6394#issuecomment-656193666 for a workaround.

**Comment by StephenBarlow** [comment]
Green to go for tomorrow! I'll try to get some eyes on the preview docs today, and we can of course follow up quickly with any hiccups folks encounter.

**Comment by hwillson** [comment]
Awesome @StephenBarlow - I'll get this merged first thing in the AM. Thanks very much!

**Comment by StephenBarlow** [comment]
@hwillson Hold off on merging tomorrow until our esteemed colleague @trevorblades is prepared to push a complementary Netlify change roughly simultaneously. Otherwise, all of the existing 3.0-beta doc URLs out there will be broken for a time.

**Pull request merged**